### PR TITLE
Considering upgrading to Junit5

### PR DESCRIPTION
**Pull request by renoth**
Proof of concept of the upgrade to Junit5:

Thanks for the quick reply, i supplied the changes to WicketTester to remove the junit4 dependencies, i think TagTester and Formtester also have junit4 dependencies, ideally Junit would be removed as a dependency altogether here and some more light-weight assertion framework could be used, maybe assertj or hamcrest.

 i also did the migration to Junit5 in the wicket-util module to show how the testcode would change. i commented out some tedious refactoring classes, the remaining tests should run.

As for the pom.xml changes i am not sure if these are correct (especially in wicket-util)

The removal of Junit as a dependency to WicketTester could also be considered if it is acceptable to jave AssertJ or Hamcrest  or another validation library as dependency.

### CODE REVIEW

File: wicket-core/src/main/java/org/apache/wicket/util/tester/WicketTester.java [hunk 1]
```diff
@@ -54,14 +38,24 @@
...
 import org.slf4j.LoggerFactory;
 
+import javax.servlet.ServletContext;
+import java.io.Serializable;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Locale;
+
+import static org.junit.jupiter.api.Assertions.*;
```
Comment (solomax) [inline wicket-core/src/main/java/org/apache/wicket/util/tester/WicketTester.java]: Whild card imports should be expanded

Lot's of re-formatting here :( hard to get the changes ....

File: wicket-util/src/main/java/org/apache/wicket/util/markup/xhtml/WellFormedXmlTestCase.java [hunk 2]
```diff
@@ -16,23 +16,18 @@
  */
 package org.apache.wicket.util.markup.xhtml;
 
+import org.junit.jupiter.api.Test;
+import org.xml.sax.*;
```
Comment (solomax) [inline wicket-util/src/main/java/org/apache/wicket/util/markup/xhtml/WellFormedXmlTestCase.java]: Another wild-card import

File: wicket-util/src/test/java/org/apache/wicket/util/collections/IntHashMapTest.java [hunk 3]
```diff
@@ -48,9 +46,9 @@ public void serialize() throws IOException, ClassNotFoundException
 		ByteArrayInputStream bais = new ByteArrayInputStream(serialized);
 		ObjectInputStream ois = new ObjectInputStream(bais);
 		IntHashMap<String> deserialized = (IntHashMap<String>) ois.readObject();
-		assertThat(deserialized, is(notNullValue()));
-		assertThat(deserialized.get(1), is(equalTo("one")));
-		assertThat(deserialized.get(2), is(equalTo("two")));
+		assertNotNull(deserialized);
+		assertEquals(deserialized.get(1), "one");
```
Comment (solomax) [inline wicket-util/src/test/java/org/apache/wicket/util/collections/IntHashMapTest.java]: Is this correct parameter order?

File: wicket-core/src/main/java/org/apache/wicket/util/tester/WicketTester.java (hunk 1 above)
Comment (renoth) [reply wicket-core/src/main/java/org/apache/wicket/util/tester/WicketTester.java]: Sorry, i will try to properly adjust the autoformatter

**Comment by martin-g** [comment]
AFAIU this PR replaces JUnit 4.x with JUnit 5.x. Am I correct ?
I've seen usage of opentest4j classes which I guess come as transitive dependency of JUnit 5.x.
I am fine with such kind of change in `master` branch (Wicket 9.x). But I'm afraid we cannot do this change in 8.x.

**Comment by renoth** [comment]
Yes this is correct, this PR is not intended to be merged soon or at all but just as a demo on what the code changes would be if upgrading to Junit5 and removing Junit4 dependencies from WicketTester. 
I would be happy to supply changes to the Wicket 9 branch if you want to.

**Comment by solomax** [comment]
+1 from me :)

**Comment by svenmeier** [comment]
+1 from me too

**Comment by martin-g** [comment]
@renoth We have an agreement to upgrade to JUnit 5.x. You can continue with the work on this PR! Please also open a ticket in JIRA for the changelog. Thank you!

**Comment by renoth** [comment]
I have returned from vacation and created https://issues.apache.org/jira/browse/WICKET-6583 i will begin the upgrade over the next weeks. Comments and suggestions very welcome.

File: wicket-util/src/test/java/org/apache/wicket/util/collections/IntHashMapTest.java (hunk 3 above)
Comment (renoth) [reply wicket-util/src/test/java/org/apache/wicket/util/collections/IntHashMapTest.java]: No you are right, it should be reversed (expected, actual)

File: wicket-core/src/main/java/org/apache/wicket/util/tester/WicketTester.java [hunk 4]
```diff
@@ -980,7 +976,7 @@ public void executeListener(final Class<?> testClass, final Component component,
 	public void executeBehavior(final Class<?> testClass, final AbstractAjaxBehavior behavior,
 		final String filename) throws Exception
 	{
-		assertNotNull(behavior);
+		assert(behavior != null);
```
Comment (martin-g) [inline wicket-core/src/main/java/org/apache/wicket/util/tester/WicketTester.java]: This looks wrong.

File: wicket-native-websocket/wicket-native-websocket-core/pom.xml [hunk 5]
```diff
@@ -37,8 +37,9 @@
         </dependency>
 
         <dependency>
-            <groupId>junit</groupId>
-            <artifactId>junit</artifactId>
+            <groupId>org.junit.jupiter</groupId>
+            <artifactId>junit-jupiter-engine</artifactId>
+            <scope>test</scope>
```
Comment (martin-g) [inline wicket-native-websocket/wicket-native-websocket-core/pom.xml]: `scope` must be in the parent's `dependencyManagement`

File: wicket-util/src/main/java/org/apache/wicket/util/markup/xhtml/WellFormedXmlTestCase.java (hunk 2 above)
Comment (renoth) [reply wicket-util/src/main/java/org/apache/wicket/util/markup/xhtml/WellFormedXmlTestCase.java]: fixed

File: wicket-core/src/main/java/org/apache/wicket/util/tester/WicketTester.java (hunk 1 above)
Comment (renoth) [reply wicket-core/src/main/java/org/apache/wicket/util/tester/WicketTester.java]: i applied the autoformatter

**Comment by martin-g** [comment]
I'm trying to merge this PR.

**Comment by martin-g** [comment]
Merged to master!

Thank you for the work, Johannes!

**Comment by martin-g** [comment]
Please close the PR.

**Comment by renoth** [comment]
No Problem, thanks for accepting the PR.

**Pull request closed without merging**